### PR TITLE
Added Validation to Scaffold

### PR DESCRIPTION
**Pull request by dbpolito**

### CODE REVIEW

File: views/scaffolding/orm/actions/edit.php [hunk 1]
```diff
@@ -2,20 +2,35 @@
...
+				else
+				{
+					Session::set_flash('error', 'Could not update <?php echo $singular_name; ?> #' . $id);
+				}
+			}
 			else
 			{
-				Session::set_flash('error', 'Could not update <?php echo $singular_name; ?> #' . $id);
+<?php foreach ($fields as $field): ?>
+				$<?php echo $singular_name; ?>-><?php echo $field['name']; ?> = Input::post('<?php echo $field['name']; ?>');
```
Comment (philsturgeon) [inline views/scaffolding/orm/actions/edit.php]: Would it be better to use `$val->validated()` to give the array back instead? Less verbose but neater. Not sure.

File: views/scaffolding/orm/actions/edit.php [hunk 2]
```diff
@@ -2,20 +2,35 @@
 
 		if (Input::method() == 'POST')
 		{
+			$val = Model_<?php echo $model_name; ?>::validate('edit');
+			
+			if ($val->run())
+			{
 <?php foreach ($fields as $field): ?>
```
Comment (philsturgeon) [inline views/scaffolding/orm/actions/edit.php:6]: I notice this is being done twice, just do it inside the `if (Input::method() == 'POST')`?

**Comment by dbpolito** [comment]
The $val->validated() returns an array and $<?php echo $singular_name; ?> is an object, maybe i'm being an idiot and forgot how concat an array to an object! :(

**Comment by philsturgeon** [comment]
Sorry bad wording. Use `$val->validated($field['name'])` instead of `Input::post($field['name'])`, so if the validation rules modify the value it will be kept.

**Comment by dbpolito** [comment]
Done. Anything else?

**Comment by okhayat** [comment]
As mentioned in the pull request, it would be great if we can set rules to make sure an integer field is numeric or integer, set the max_length for char, varchar, and string fields, and maybe other similar things, like email -> valid_email ... etc

**Comment by dbpolito** [comment]
Ok, coming soon! :D

**Comment by dbpolito** [comment]
Thoughts?

File: classes/generate/scaffold.php [hunk 3]
```diff
@@ -68,15 +68,15 @@ public static function forge($args, $subfolder)
...
+
 		// Replace / with _ and classify the rest. DO NOT singularize
 		$controller_name = \Inflector::classify(static::$controller_prefix.str_replace(DS, '_', $name), false);
-		
+
 		// Replace / with _ and classify the rest. Singularize
 		$model_name = \Inflector::classify(static::$model_prefix.str_replace(DS, '_', $name));
 
 		// Either foo or folder/foo
-		$view_path = $controller_path = str_replace(
```
Comment (philsturgeon) [inline classes/generate/scaffold.php:79]: Why was $view_path removed?

File: views/scaffolding/orm/model.php [hunk 4]
```diff
@@ -28,4 +28,37 @@ class Model_<?php echo $model_name; ?> extends Model
...
+		$val = Validation::forge($factory);
+
+<?php foreach ($fields as $field): ?>
+<?php
+		$rules = array('required');
+		
+		if ($field['name'] === 'email')
+		{
+			$rules[] = 'valid_email';
+			$rules[] = !is_null($field['constraint']) ? "max_length[{$field['constraint']}]" : 'max_length[255]';
```
Comment (philsturgeon) [inline views/scaffolding/orm/model.php]: Could you make `!is_null(` into `! is_null(` please? Style-guide rule. :)

**Comment by dbpolito** [comment]
I have no idea why $view_path was removed, probably it was an automatic merge before the pull request. Anyway, now it seems to be right. Anything let me know.

**Comment by okhayat** [comment]
Yes, my bad. nothing more for text field

**Comment by okhayat** [comment]
Oh, maybe text fields (varchar, string, text) need a 'trim' also.
Sorry :)

**Comment by philsturgeon** [comment]
Let's not make anymore assumptions than we have to, we are not trying to build their application for them. I'm already uncertain if we should be requiring fields by default.

I generally prefer to do less and let people add things in, that to make people take things out to work properly. Thats kinda the basis for everything oil does.

**Comment by okhayat** [comment]
Ok. I've tested the code locally and it works great. Do you think it's good to go as is then?

**Pull request merged**